spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -510,22 +505,14 @@ function chanUpgradeTry(
         (currentTimestamp() > timeout.timeoutTimestamp && timeout.timeoutTimestamp != 0)
     )
 
-    // if OPEN, then initialize handshake with upgradeFields
-    // otherwise, assert that the upgrade fields are the same for crossing-hellos case
-    if currentChannel.state == OPEN {
-        // if the counterparty sequence is greater than the current sequence, we fast forward to the counterparty sequence
-        // so that both channel ends are using the same sequence for the current upgrade
-        // initUpgradeChannelHandshake will increment the sequence so after that call
-        // both sides will have the same upgradeSequence
-        if counterpartyUpgradeSequence > currentChannel.upgradeSequence {
-            currentChannel.upgradeSequence = counterpartyUpgradeSequence - 1
-        }
+    // if the counterparty sequence is greater than the current sequence, we fast forward to the counterparty sequence
+    // so that both channel ends are using the same sequence for the current upgrade
+    // initUpgradeChannelHandshake will increment the sequence so after that call
+    // both sides will have the same upgradeSequence
+    abortTransactionUnless(counterpartyUpgradeSequence > currentChannel.upgradeSequence)
 
-        initUpgradeChannelHandshake(portIdentifier, channelIdentifier, upgradeFields, counterpartyUpgrade.timeout)
-    } else if currentChannel.state == INITUPGRADE {
-        existingUpgrade = publicStore.get(channelUpgradePath)
-        abortTransactionUnless(existingUpgrade.fields == upgradeFields)
-    }
+    upgrade = publicStore.get(channelUpgradePath)
+    abortTransactionUnless(upgrade.fields == counterpartyUpgrade.fields)
```

Comment:
Is this check necessary? Seems like a mistake - counterpartyUpgrade fields may have a different connection ID to upgrade fields connection ID and that should be valid, right? i.e. both connection ends do not need to share the same ID so we cannot make this assertion

Reply:
Yes you're right. The necessary checks are made in `startFlushHandshake`